Infrastructure/Extentions/ExceptionMiddlewareExtensions.cs

```
@@ -0,0 +1,179 @@
+﻿using Application.Contracts;
+using Application.Exceptions;
+using Domain.Entities;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Domain.Entities;
+using Application.Exceptions;
+using Microsoft.Extensions.Localization;
+using Domain;
+using Application.Resources;
+using Application.Models;
+
+namespace Infrastructure.Extentions {
+    public class ExceptionMiddlewareExtensions
+    {
+        private readonly RequestDelegate _next;
+        private readonly IStringLocalizer<LocalizationResource> _localizer;
+
+        public ExceptionMiddlewareExtensions(RequestDelegate next, IStringLocalizer<LocalizationResource> localizer)
+        {
+            _next = next;
+            _localizer = localizer;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionAsync(context, ex, _localizer);
+            }
+        }
+
+        private static Task HandleExceptionAsync(HttpContext context, Exception exception, IStringLocalizer localizer)
+        {
+
+            HttpStatusCode statusCode;
+            List<LocalizedString> errorMessages = new List<LocalizedString>();
+
+           if (exception is FluentValidationException fluentValidationException)
+            {
+                statusCode = HttpStatusCode.BadRequest;
+                foreach (var errorMessage in fluentValidationException.ErrorMessages)
+                {
+                    errorMessages.Add(localizer[errorMessage]);
+                }
+            }
+            else if (exception is EmailInUseException emailInUseException)
```

Instead of listing individual exception validations, you can encapsulate them under broader exception types like `CustomException` or `BadRequestException`. This way, when you need to throw an exception, you can use one of these types and provide a clear message indicating the reason for the failure. This approach helps centralize exception handling and makes your code more organized.